docs/isl-2-0/spec.md

```diff
@@ -0,0 +1,673 @@
+Number-typed `null` values are never valid for any number range.
+
+A `timestamp` range includes any `timestamp` that falls between the minimum and maximum in chronological order.
+All [`timestamp` values](https://amzn.github.io/ion-docs/docs/spec.html#timestamp) represent an instant in time.
+A `timestamp` with limited precision (for example, a year-only timestamp like `2007T`) maps to an instant by assuming that all unspecified time unit fields are effectively zero (or one for the month and day units). 
+`2007T` would map to the instant represented by `2007-01-01T00:00.000-00:00`.
+Note that `timestamp` values that do not have a time component (that is: `YYYYT`, `YYYY-MMT`, and `YYYY-MM-DDT`) timestamps) have an unknown offset (`-00:00`).
+Timestamps that have an unknown offset are UTC timestamps that make no assertion about the offset in which they occurred.
+The value `null.timestamp` is never in the bounds of a timestamp range, and `timestamp` ranges do not include values of any other type.
+
+# Open Content
+
+{% include grammar-element.md productions="user_content_declaration,user_content_declaration_field" %}
+
+The Ion Schema 2.0 specification allows users to insert additional content into a schema document that is not part of the Ion Schema Language.
+This additional content may be used for documentation, integrations with other tools or frameworks, or any other desired purpose.
+
+A schema header, type definition, or schema footer may include extra fields that are not explicitly stated in the Ion Schema specification.
+These extra fields may have any field name that is not a *keyword*.
+If the field name is a *reserved symbol*, its use as open content must be declared in the appropriate subfield of the `user_content` field in the schema header.
+For example, the reserved word `list_type` may be used in a type definition if `user_content: { type: [ list_type ] }` is present in the schema header.
+
+An Ion Schema MAY include extra top-level values that are not explicitly specified in the Ion Schema specification, but any top-level open content MUST NOT be annotated with a *reserved symbol*.
+Note that Ion Schema version markers are always interpreted as Ion Schema version markers and can never be valid open content.
+
+No other open content is allowed except for what is explicitly identified in this section.
+
+### Reserved Symbols
+
+The set of reserved symbols SHALL be all symbols matching the regular expression `^(\$ion_schema(_.*)?|[a-z][a-z0-9]*(_[a-z0-9]+)*)$`.
+Informally stated, this is the symbol `$ion_schema`, all symbols starting with `$ion_schema_`, and all [identifier symbols](https://amzn.github.io/ion-docs/docs/spec.html#symbol) that are *snake case* and start with an unaccented ascii, lower-case letter.
+
+### Keywords
+A keyword is a reserved symbol that has been assigned a meaning by the Ion Schema specification.
+Whether a reserved symbol is considered a keyword is context dependent.
```

Comment:
What do we buy ourselves by making keyword status context dependent rather than saying (e.g.) `imports` is a keyword everywhere? I can't think of any programming languages or similar that have context-sensitive keyword reservations.

Reply:
Presumably whoever wrote the RFC for open content thought that the set of keywords should be as minimal as possible, and that's why it's context dependent.

What do we gain by it? Nothing, really. There's no benefit for Ion Schema users except that they could use "valid_values", etc. in the header or footer (not a compelling reason, IMO). It increases the cognitive burden for users, and it (slightly) complicates implementations of Ion Schema.

Reply:
I'm a big fan of not picking up cognitive overhead and implementation complication without stronger motivation. Good catch @zslayton and good explanation @popematt. I'm in favor of changing this to remove context dependence.